front/lib/api/assistant/generation.ts

```diff
@@ -145,6 +147,32 @@ export async function renderConversationForModel({
       return new Err(new Error(`Error tokenizing model message: ${err}`));
     }
   }
+  async function tokenSplit(
```

Comment:
white space above

Reply:
Should this be in types/ ?

Reply:
Next to safeSubstring?

Reply:
if you mean the tokenSplit function, i'd say it's coherent to have it here next to tokenCount, similar methods, both only used here

moving them to types could make sense if they're used elsewhere, but not next to `safeSubstring` which is a purely string util (and is in "string_utils") not related to tokenization